internal/dcache/rpc/server/handler.go

```diff
@@ -1423,27 +1659,60 @@ func (h *ChunkServiceHandler) UpdateMV(ctx context.Context, req *models.UpdateMV
 		return nil, rpc.NewResponseError(models.ErrorCode_InvalidRequest, errStr)
 	}
 
-	rvInfo := h.getRVInfoFromRVName(req.RVName)
-	if rvInfo == nil {
-		log.Err("ChunkServiceHandler::UpdateMV: Invalid RV %s", req.RVName)
-		return nil, rpc.NewResponseError(models.ErrorCode_InvalidRV, fmt.Sprintf("invalid RV %s", req.RVName))
-	}
+	clustermapRefreshed := false
+	for {
+		rvInfo := h.getRVInfoFromRVName(req.RVName)
+		if rvInfo == nil {
+			errStr := fmt.Sprintf("node %s does not host %s", rpc.GetMyNodeUUID(), req.RVName)
+			log.Err("ChunkServiceHandler::UpdateMV: %s", errStr)
+			common.Assert(false, errStr)
+			return nil, rpc.NewResponseError(models.ErrorCode_InvalidRV, errStr)
+		}
 
-	mvInfo := rvInfo.getMVInfo(req.MV)
-	if mvInfo == nil {
-		errStr := fmt.Sprintf("RV %s is not member of MV %s", req.RVName, req.MV)
-		log.Err("ChunkServiceHandler::UpdateMV: %s", errStr)
-		return nil, rpc.NewResponseError(models.ErrorCode_InvalidRequest, errStr)
-	}
+		//
+		// A membership update RPC is only sent to RVs which are already members of the MV, and it is sent
+		// when the membership changes (an existing RV is replaced by another RV by the fix-mv workflow).
+		// Since the sender is referring to the global clustermap and this RV is part of the given MV as
+		// per the global clustermap, since an RV is added to an MV only after a successful JoinMV response
+		// from all component RVs, we *must* have the MV replica in our rvInfo.
+		//
+		mvInfo := rvInfo.getMVInfo(req.MV)
+		if mvInfo == nil {
+			errStr := fmt.Sprintf("%s is not part of %s", req.RVName, req.MV)
+			log.Err("ChunkServiceHandler::UpdateMV: %s", errStr)
+			common.Assert(false, errStr)
+			return nil, rpc.NewResponseError(models.ErrorCode_InvalidRequest, errStr)
+		}
```

Comment:
`rvInfo` and `mvInfo` checks can be moved outside of the for loop.

Reply:
Actually I had deliberately included it inside the loop, since we are updating info from clustermap and clustermap can technically result in the rv being removed from the mv. Right now we may not be using it as such.